### PR TITLE
Disallow negative duration for animation and transition

### LINKED ISSUE #15343: transition-duration should not accept negative time value
Spec: https://drafts.csswg.org/css-transitions/#transition-duration-property

Relevant code: https://github.com/servo/servo/blob/master/components/style/properties/longhand/box.mako.rs#L465

We probably need `Time::parse_non_negative` as well.

### PR DESCRIPTION
**Pull request by hiikezoe**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15343

- [X] There are tests for these changes written by @simon-whitehead . Thank you!

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16829)

### CODE REVIEW

**Comment by hiikezoe** [comment]
oh-oh.

**Comment by hiikezoe** [comment]
OK, now test result is good. I will ask to review tomorrow since homu now is now stuck.

**Comment by hiikezoe** [comment]
This is the inherited PR of @simon-whitehead 's work (#15658).

r? @emilio

File: components/style/properties/longhand/box.mako.rs [hunk 1]
```diff
@@ -824,7 +824,11 @@ ${helpers.single_keyword("overflow-x", "visible hidden scroll auto",
     pub use properties::longhands::transition_duration::single_value::SpecifiedValue;
     pub use properties::longhands::transition_duration::single_value::computed_value;
     pub use properties::longhands::transition_duration::single_value::{get_initial_value, get_initial_specified_value};
-    pub use properties::longhands::transition_duration::single_value::parse;
+
+    pub fn parse(context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue,()> {
```
Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: nit: space after comma.

**Review by emilio — APPROVED** [review]
r=me, with that nit.

**Pull request merged**